### PR TITLE
docs: Update version_notes.adoc with 0.13.1 release notes

### PR DESCRIPTION
**Pull request by odednaor**
### Description of the Changes

Release notes taken from https://github.com/starkware-libs/cairo-lang/releases/tag/v0.13.1a0.

### PR Preview URL

[Release notes](https://starknet-io.github.io/starknet-docs/pr-1160/documentation/starknet_versions/version_notes/)

### Check List

- [x] Changes have been done against main branch, and PR does not conflict
- [x] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

### CODE REVIEW

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 1]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
 |Goerli Testnet (deprecated)|0.13.0|1.4.0|2.0.0 - 2.5.4
 |===
 
+
+[id="version0.13.1"]
+== Starknet v0.13.1
+
+* *Cheaper data availability:* Starknet uses link:https://eips.ethereum.org/EIPS/eip-4844[EIP-4844].
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:28]: Why is this section not considered _Technical changes_?

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 2]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+* Time-related syscalls when called from account contract's `__validate__` function:
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:29]: What about validate_deploy and validate_declare?

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 3]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+** `timestamp` returns the hour, rounded down.
+** `block_number` returns the block number, rounded down to the nearest multiple of 100.
+* *Optimization:* Load only the used functions in a contract into memory when generating the proof
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:32]: Is this a sequencer change? Or something else? Please specify.

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 4]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+* Add `starknet-compiled-class-hash` command
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:33]: Add this command to what?

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 5]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+
+=== API changes:
+
+* Starknet block:
+** New field `l1_da_mode`, which indicates whether EIP-4844 was used in the block.
+** `eth_l1_gas_price` and `strk_l1_gas_price` were replaced, and the information now contains the data gas price (EIP 4844) in addition to the regular gas price
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:39]:
```suggestion
** `eth_l1_gas_price` and `strk_l1_gas_price` were replaced, and the information now contains the data gas price (EIP-4844) in addition to the regular gas price
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 6]
```diff
@@ -21,8 +21,35 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+** `execution_resources` in the transaction receipt now contains:
+*** data availability resources
+*** syscall resources, which contribute to the transaction fee but were not included in the receipt before this version.
+** `transaction_commitment` and `event_commitment` are added to the block (zero is returned for old blocks)
+
+=== Technical changes:
+
+* Starknet now supports multiple L1 providers.
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:47]:
```suggestion
Starknet now supports multiple L1 providers.
```
One item doesn't get a bullet. Only two or more items.

**Review by stoobie — CHANGES_REQUESTED** [review]
Several questions and suggestions.

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc (hunk 2 above)
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:29]:
```suggestion
* Time-related syscalls when called from an account contract's `__validate__` function:
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc (hunk 3 above)
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:32]:
```suggestion
* *Optimization:* Load only the used functions in a contract into memory when generating the proof.
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc (hunk 4 above)
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:33]:
```suggestion
* Add `starknet-compiled-class-hash` command.
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 7]
```diff
@@ -265,7 +292,7 @@ Technical changes:
 
 [id="version0.9.1"]
 == Starknet v0.9.1 (July 20, 22)
-
+November
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:295]:
```suggestion
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 8]
```diff
@@ -16,13 +16,40 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
-
-|Sepolia Testnet|0.13.0|1.4.0|2.0.0 - 2.5.4
+|Sepolia Testnet|0.13.1|1.5.0|2.0.0 - 2.6
 |Goerli Testnet (deprecated)|0.13.0|1.4.0|2.0.0 - 2.5.4
 |===
 
+[id="version0.13.1"]
+== Starknet v0.13.1
+
+=== New features and enhancements
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:26]:
```suggestion
=== New features and enhancements

```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 9]
```diff
@@ -16,13 +16,41 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:26]:
```suggestion
[discrete]
=== New features and enhancements
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 10]
```diff
@@ -16,13 +16,44 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+== Starknet v0.13.1
+
+[discrete]
+=== New features and enhancements
+
+* *Cheaper data availability:* Starknet uses link:https://eips.ethereum.org/EIPS/eip-4844[EIP-4844].
+* Time-related syscalls when called from an account contract's `__validate__` function:
+** `timestamp` returns the hour, rounded down.
+** `block_number` returns the block number, rounded down to the nearest multiple of 100.
+* *Optimization:* Load only the used functions in a contract into memory when generating the proof.
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:33]:
```suggestion
* *Optimization:* Load into memory only the functions in a contract that are actually used when generating the proof.
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 11]
```diff
@@ -16,13 +16,44 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+
+[discrete]
+=== New features and enhancements
+
+* *Cheaper data availability:* Starknet uses link:https://eips.ethereum.org/EIPS/eip-4844[EIP-4844].
+* Time-related syscalls when called from an account contract's `__validate__` function:
+** `timestamp` returns the hour, rounded down.
+** `block_number` returns the block number, rounded down to the nearest multiple of 100.
+* *Optimization:* Load into memory only the functions in a contract that are actually used when generating the proof.
+* Add `starknet-compiled-class-hash` command.
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:34]: @ArielElp Add it to what?

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 12]
```diff
@@ -16,13 +16,44 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
 |===
 
+[id="version0.13.1"]
+== Starknet v0.13.1
+
+[discrete]
+=== New features and enhancements
+
+* *Cheaper data availability:* Starknet uses link:https://eips.ethereum.org/EIPS/eip-4844[EIP-4844].
+* Time-related syscalls when called from an account contract's `__validate__` function:
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:30]: Does this also apply to `__validate_declare__` and `__validate_deploy__`?

Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:30]:
```suggestion
* Time-related syscalls when called from an account contract's `+__validate__+` function:
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc [hunk 13]
```diff
@@ -16,13 +16,44 @@ include::ROOT:partial$snippet_important_goerli2_removed.adoc[]
...
+
+* New field `l1_da_mode`, which indicates whether EIP-4844 was used in the block.
+* `eth_l1_gas_price` and `strk_l1_gas_price` were replaced, and the information now contains the data gas price (EIP-4844) in addition to the regular gas price
+* `execution_resources` in the transaction receipt now contains:
+** data availability resources
+** syscall resources, which contribute to the transaction fee but were not included in the receipt before this version.
+* `transaction_commitment` and `event_commitment` are added to the block (zero is returned for old blocks)
+
+[discrete]
+=== Technical changes:
```
Comment (stoobie) [inline components/Starknet/modules/starknet_versions/pages/version_notes.adoc:49]:
```suggestion
=== Infrastructure updates
```

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc (hunk 11 above)
Comment (stoobie) [reply components/Starknet/modules/starknet_versions/pages/version_notes.adoc:34]: I'm removing this line. Lior explained that this is something specific to cairo-lang that he added to the cairo-lang repo, but it is not useful here.

File: components/Starknet/modules/starknet_versions/pages/version_notes.adoc (hunk 4 above)
Comment (stoobie) [reply components/Starknet/modules/starknet_versions/pages/version_notes.adoc]: Lior explained that this is a helper utility for CairoZero, so I removed this.

**Comment by iliav-starkware** [comment]
LGTM

**Review by stoobie — APPROVED** [review]

**Review by iliav-starkware — APPROVED** [review]

**Pull request merged**